msg.go

```diff
@@ -11,15 +11,32 @@ package dns
 //go:generate go run msg_generate.go
 
 import (
+	crand "crypto/rand"
+	"encoding/binary"
 	"encoding/hex"
+	"log"
 	"math/big"
 	"math/rand"
 	"net"
 	"reflect"
 	"strconv"
-	"time"
 )
 
+func init() {
+	// Initialize default math/rand source using crypto/rand to provide better
+	// security without the performance trade-off.
+	buf := make([]byte, 8)
+	_, err := crand.Read(buf)
+	if err != nil {
+		// Failed to read from cryptographic source, warn user and
+		// fallback to default initial seed (1) by returning early
+		log.Printf("Failed to seed math/rand with crypto/rand.Read for message ID generation: %s\n", err)
```

Comment:
prolly nobody ever will read this and the random is no worse then now, so I would just drop the log - leave the comment though!